app/lib/preload.js

```diff
@@ -9,29 +9,170 @@
  */
 
 const {
-  remote,
+  contextBridge,
   ipcRenderer
 } = require('electron');
 
-const {
-  process,
-  app
-} = remote;
+const generateId = require('./util/generate-id');
```

Comment:
Just a comment: I'm honestly surprised this works, without bundling.

Reply:
Maybe it's some electron builder magic, dunno.

Reply:
It's not any magic. We simply pack the whole app excluding tests into an asar archive:

![image](https://user-images.githubusercontent.com/28307541/115555659-095d8600-a2b0-11eb-946e-9ac0a1fc5284.png)

Reply:
Still, this implies that the pre-load happens in NodeJS? `require` is not available in the browser context.

Reply:
Preload happens in the browser context but with some special powers: https://www.electronjs.org/docs/api/sandbox-option#preload

Reply:
This is where the magic is coming from. Good find on the App.asar by the way. Maybe we can consider to not ship all these scripts some day in the future.